### PR TITLE
Swap

### PR DESCRIPTION
**Pull request by juan-cortes**
image

The current version on this PR should be feature complete. Let me clarify why I say _should_ and not _is_. As of 14th of August I've tested both an ETH to 0x swap and 0x to ETH swaps and they both completed. The rest of the functionality seems to also be working. There are, however, also a few things that will need to be tweaked and I'm sure once QA gets involved there will be many bugs, polishes and changes requested. 

Please refer to the Confluece doc for more details.

### Ready for testing
- KYC splash-screen 
- IP check for accessing the feature
- Applist connection checking necessary apps
- Connection flow to reach the swap form, selectors should reflect state of the currency
- Provided rates for a given exchange, rate range checking
- Swap functionality with error handling in case of bad rates, rejection, etc
- History of swapped operations with status support

### Known issues
- Rates are not expired on the amount step
- The export swap history functionality is **not implemented**
- Swaps from ERC20 tokens do **not** seem to reach the swap history, probably a bad implementation in the `addToSwapHistory` logic when it comes to `subOperation`.

### CODE REVIEW

File: src/screens/Swap/Connect.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+const Connect = ({ setResult }: { setResult: (result: any) => void }) => {
+  const [device, setDevice] = useState(null);
+  return (
+    <>
+      <LText semiBold style={styles.selectDevice}>
+        <Trans i18nKey={"transfer.swap.selectDevice"} />
+      </LText>
+      <SelectDevice onSelect={setDevice} autoSelectOnAdd />
+      {device ? (
+        <DeviceActionModal
```
Comment (JunichiSugiura) [inline src/screens/Swap/Connect.js:24]: Provide `onClose` prop here.
detail: https://ledgerhq.atlassian.net/browse/LL-3462

```
onClose={() => setDevice()}
```

File: src/screens/Swap/Connect.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+const Connect = ({ setResult }: { setResult: (result: any) => void }) => {
+  const [device, setDevice] = useState(null);
+  return (
+    <>
+      <LText semiBold style={styles.selectDevice}>
+        <Trans i18nKey={"transfer.swap.selectDevice"} />
+      </LText>
+      <SelectDevice onSelect={setDevice} autoSelectOnAdd />
+      {device ? (
```
Comment (JunichiSugiura) [inline src/screens/Swap/Connect.js:23]: It's handled by DeviceActionModal internally

Comment (juan-cortes) [reply src/screens/Swap/Connect.js:23]: Fair enough

File: src/screens/Swap/Connect.js (hunk 1 above)
Comment (juan-cortes) [reply src/screens/Swap/Connect.js:24]: Nice, cheers

**Review by gre — COMMENTED** [review]
(some eslint issues)

File: src/index.js [hunk 3]
```diff
@@ -75,8 +76,11 @@ type AppProps = {
 };
 
 function App({ importDataString }: AppProps) {
+  const dispatch = useDispatch();
   useAppStateListener();
 
+  getProviders().then(providers => dispatch(setSwapProviders(providers)));
```
Comment (gre) [inline src/index.js:82]: This needs to happen on an effect. Not in render.

**Review by gre — APPROVED** [review]

**Pull request merged**